src/main/java/com/kilopo/kosshop/DAO/BaseDAO.java

```diff
@@ -7,4 +7,5 @@
     T addOrUpdate(T value);
     void delete(Long id);
     List<T> getAll();
+    T getByValue(String name,Object value);
```

Comment:
i think it must be getByColumnNameAndValue.
And why it return only one object ?